### DISCUSSION TITLE
Bosca Ceoil goes BLUE (and remains open source)!

### DISCUSSION THREAD

**Discussion by YuriSizov** [YuriSizov/boscaceoil-blue#2] [Announcements]
Hey there!

For the last few months I've been [hard at work porting](https://www.patreon.com/posts/how-to-rewrite-102819465) a software synthesizer [SiON](https://github.com/keim/SiON) from ActionScript and Flash to Godot. But that wasn't the end goal of my adventure. My actual plan was to lay down the ground work for something quite special to me. And now it is time to reveal...

<p align="center">
<img width="353" alt="The title bar logo of the app" src="https://github.com/YuriSizov/boscaceoil-blue/assets/11782833/ab2616c0-0488-4270-a7c2-c4da54490b35">
</p>

**Bosca Ceoil: The Blue Album** is a complete rework of a very good and somewhat old music creation [app by Terry Cavanagh](https://boscaceoil.net/). It small, intuitive, and very fun to use — especially if you're not a savvy musician. Which was me a couple of years ago, when I discovered the original app and used it to create music and sound effects for my game, [Voxurbis: Age of Politicians](https://yurisizov.itch.io/voxurbis).

I enjoyed it so much, I wanted to help improve it and fix some of its bugs... only to discover it's an Adobe AIR (Flash) application, and working with that would be a tall order. Because of the death of the Flash platform, it became harder and harder to use the app for many users. But I really wanted to preserve this tool, so I set on a mission to revive it with a port. And as a platform I decided to use the [Godot game engine](https://godotengine.org/), which I had a lot of experience with.

Fast forward almost 2 years, and I finally have an opportunity to work on such a massive undertaking. But I believe it was worth it!

Starting today the development goes completely open source, as I finish up the features. And you can already give the project a try. All you need is the [latest official build of Godot 4.3](https://godotengine.org/download/archive/#4.3) and the [latest build of GDSiON](https://github.com/YuriSizov/gdsion/releases).

Clone or download the ZIP of the project, extract _GDSiON_ into the root folder, and use the Godot editor to launch _Bosca Ceoil Blue_! (Maybe consider contributing too, if you'd like 🙃)

<img width="691" alt="Godot_v4 3-dev5_win64_2024-04-26_14-42-28" src="https://github.com/YuriSizov/boscaceoil-blue/assets/11782833/f142c4db-b0ee-41fb-bde5-6ccaa082b668">

## Your support

My mission is to create free and open tools, just like _Bosca Ceoil Blue_ and _GDSiON_, for developers and creators like yourself. And I cannot do it without your help. Please consider [supporting me and this project on Patreon](https://patreon.com/YuriSizov), even if just a little! Thank you <3

**Comment by pYMxlobo** [comment]
is it possible to run it outside of the editor as just a normal executable?

**Reply by YuriSizov** [reply]
Sure, you can export it as an executable. The work is not complete yet, so there aren't prebuilt executables. But there will be — closer to the stable release!
